src/services/resource.js

```diff
@@ -381,7 +381,7 @@ const filterResourcesForSync = (
         .filter((rff) => rff.genericId === resource.genericId
         && rff.resourceId === resource.resourceId);
       const isReduced = matchingFromFields
-        .filter((mff) => resource.sourceFields
+        .filter((mff) => (resource.sourceFields || [])
```

Comment:
don't think we should throw this change in a random PR, restarting the backend generally resolves it so I'm not sure its a code error